lock-impl/src/test/java/com/palantir/lock/impl/AdjustableBackgroundTaskTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.lock.SimpleTimeDuration;
+import com.palantir.lock.TimeDuration;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.refreshable.SettableRefreshable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.Test;
+
+public class AdjustableBackgroundTaskTest {
```

Comment:
Tests look good! I think the only thing we're missing is validating that we only run the check every second; the current test only tests that we don't run the runnable.

Reply:
You actually called out a bug as the minimum duration of 1 second was only meant for the suppliers anyways